File: Check_if_frequencies_can_be_equal.java
class Solution {
    boolean sameFreq(String s) {
        // code here
        int[] temp=new int[26];
        for(int i=0;i<s.length();i++){
            temp[s.charAt(i)-'a']++;
        }
        HashSet<Integer> set=new HashSet<>();
        int min=Integer.MAX_VALUE;
        int max=-1;
        for(int i=0;i<26;i++){
            if(temp[i]!=0)
            set.add(temp[i]);
            if(min>temp[i] && temp[i]!=0){
                min=temp[i];
            }
            if(max<temp[i] && temp[i]!=0){
                max=temp[i];
            }
        }
        if(set.size()>2){
            return false;
        }
        if(max==min){
            return true;
        }
        int count1=0,count2=0;
        for(int i=0;i<26;i++){
            if(min==temp[i]){
                count1++;
            }
            if(max==temp[i]){
                count2++;
            }
        }
        if(count1>1 && count2>1){
            return false;
        }
        if(min ==1 && count1==1){
            return true;
        }
        return (Math.abs(max-min)==1 && count2==1)? true:false;
    }
}

File: Check_if_frequencies_can_be_equal.cpp
class Solution {
  public:
    bool sameFreq(string& s) {
        vector<int> mp(26, 0);
        for(char ch : s) mp[ch - 'a']++;
        unordered_set<int> ss;
        int mx = 0, mn = 100000;
        unordered_map<int, int> freq;
        for(int i = 0 ; i < 26 ; i++) {
            if(mp[i] != 0) {
                ss.insert(mp[i]);
                freq[mp[i]]++;
                mx = max(mx, mp[i]);
                mn = min(mn, mp[i]);
            }
        }
        if(ss.size() > 2) return false;
        if((mx == mn) || (mn == 1 && freq[mn] == 1) || (freq[mx] == 1 && mx - mn == 1)) return true;
        return false;
    }
};
